superset/db_engine_specs.py

```diff
@@ -127,8 +126,7 @@ def get_query_without_limit(cls, sql):
                 (?ix)        # case insensitive, verbose
                 \s+          # whitespace
                 LIMIT\s+\d+  # LIMIT $ROWS
-                ;?           # optional semi-colon
-                (\s|;)*$     # remove trailing spaces tabs or semicolons
+                (\s|;|.)*$   # remove the reset of the chars
```

Comment:
How about we selected everything after the limit and also passed that along to the caller so the caller can do something lie "{before_limit} {new_limit} {after_limit}" That should fix problem 3 (adding the things after the limit). And to do that, we will just change this line to to `(.*$)` and pass the regex results.

Reply:
https://github.com/apache/incubator-superset/pull/5295